test/powershell/Modules/Microsoft.PowerShell.Management/Clear-EventLog.Tests.ps1

```diff
@@ -13,20 +13,27 @@ Describe "Clear-EventLog cmdlet tests" -Tags @('CI', 'RequireAdminOnWindows') {
 
     It "should be able to Clear-EventLog" -Pending:($True) {
       Remove-EventLog -LogName TestLog -ea Ignore
-      {New-EventLog -LogName TestLog -Source TestSource -ErrorAction Stop} | Should -Not -Throw
-      {Write-EventLog -LogName TestLog -Source TestSource -Message "Test" -EventID 1 -ErrorAction Stop } | Should -Not -Throw
-      {$result=Get-EventLog -LogName TestLog}                     | Should -Not -Throw
-      ($result.Count)                                             | Should -Be 1
-      {Clear-EventLog -LogName TestLog}                           | Should -Not -Throw
-      $result=Get-EventLog -LogName TestLog -ea Ignore
-      ($result.Count)                                             | Should -Be 0
-      {Remove-EventLog -LogName TestLog -ErrorAction Stop}        | Should -Not -Throw
+      { New-EventLog -LogName TestLog -Source TestSource -ErrorAction Stop } | Should -Not -Throw
+      { Write-EventLog -LogName TestLog -Source TestSource -Message "Test" -EventID 1 -ErrorAction Stop } | Should -Not -Throw
+      { Get-EventLog -LogName TestLog }                           | Should -Not -Throw
+      $result = Get-EventLog -LogName TestLog
+      $result.Count                                               | Should -Be 1
+      { Clear-EventLog -LogName TestLog }                         | Should -Not -Throw
+      $result = Get-EventLog -LogName TestLog -ErrorAction Ignore
+      $result.Count                                               | Should -Be 0
+      { Remove-EventLog -LogName TestLog -ErrorAction Stop }      | Should -Not -Throw
     }
-    It "should throw 'The Log name 'MissingTestLog' does not exist' when asked to clear a log that does not exist" -Pending:($True) {
-      Remove-EventLog -LogName MissingTestLog -ea Ignore
-      { Clear-EventLog -LogName MissingTestLog -ErrorAction Stop } | Should -Throw -ErrorId "Microsoft.PowerShell.Commands.ClearEventLogCommand"
+
+    It "should throw 'The Log name 'MissingTestLog' does not exist' exception message when asked to clear a log that does not exist" -Pending:($True) {
+      Remove-EventLog -LogName MissingTestLog -ErrorAction Ignore
+      { Clear-EventLog -LogName MissingTestLog -ErrorAction Stop } | Should -Throw -ExpectedMessage 'The Log name "MissingTestLog" does not exist'
```

Comment:
The problem with checking ExpectedMessage is that it will fail on non en-us systems when we eventually get localization of error messages.  Looking at the Pester code, it currently doesn't check current locale.  For now, I think it's best to just remove this test.  Your updated test below on the exception seems like an improvement already.

Reply:
Thanks for the explanation. Removed.